elfo-network/src/connection.rs

```diff
@@ -331,12 +332,44 @@ struct SocketReader {
     tx_flows: Arc<TxFlows>,
     rx_flows: Arc<Mutex<RxFlows>>,
     requests: Arc<Mutex<OutgoingRequests>>,
+    handle_addr: Addr,
 }
 
 impl SocketReader {
     async fn exec(mut self) -> ConnectionClosed {
-        // TODO: error handling.
-        while let Some(network_envelope) = self.rx.recv().await.unwrap() {
+        loop {
+            let network_envelope = match self.rx.recv().await {
+                Ok(Some(envelope)) => envelope,
+                Ok(None) => break,
+                Err(ReadError::RequestSkipped(details)) => {
+                    scope::set_trace_id(details.trace_id);
+                    // If there was an error while decoding a request, we need to notify the remote
+                    // actor in order to avoid waiting for a response indefinitely.
+                    let sender = self
+                        .ctx
+                        .book()
+                        .get(self.handle_addr)
+                        .expect("bug: remote group is missing in the address book");
+                    let token = ResponseToken::new(
+                        details.sender,
+                        details.request_id,
+                        details.trace_id,
+                        self.ctx.book().clone(),
+                    );
+                    sender.respond(token, Err(RequestError::Failed));
+                    error!(
```

Comment:
The same event produces the two related errors (in the decoder and here). I think it's better to provide more information in the first place and, optionally, `debug!` here.

I think later, `Skipped` and `RequestSkipped` will be merged anyway (because we should also handle `Skipped` to fix flow control state), so for now it's ok just to replace this level with `debug` one